src/main/kotlin/no/skatteetaten/aurora/boober/service/DeployService.kt

```diff
@@ -87,22 +86,22 @@ class DeployService(
 
     private fun prepareDeployEnvironment(environment: AuroraDeployEnvironment, projectExist: Boolean): List<OpenShiftResponse> {
 
-        val affiliation = environment.affiliation
-        val namespace = environment.namespace
+        val namespaceName = environment.namespace
 
-        val createNamespaceResponse = openShiftObjectGenerator.generateProjectRequest(environment).let {
-            openShiftClient.createOpenShiftCommand(namespace, it, projectExist)
-        }.let { openShiftClient.performOpenShiftCommand(namespace, it) }
-
-        val updateNamespaceResponse = openShiftClient.createUpdateNamespaceCommand(namespace, affiliation).let {
-            openShiftClient.performOpenShiftCommand(namespace, it)
+        val responses = mutableListOf<OpenShiftResponse>()
+        if (!projectExist) {
+            val projectRequestResponse = openShiftCommandBuilder.generateProjectRequest(environment)
+                    .let { openShiftClient.performOpenShiftCommand(namespaceName, it) }
+            responses.add(projectRequestResponse)
         }
 
-        val updateRoleBindingsResponse = openShiftObjectGenerator.generateRolebindings(environment.permissions).map {
-            openShiftClient.createOpenShiftCommand(namespace, it, createNamespaceResponse.success, true)
-        }.map { openShiftClient.performOpenShiftCommand(namespace, it) }
+        val namespace = openShiftCommandBuilder.generateNamespace(environment)
+        val roleBindings = openShiftCommandBuilder.generateRolebindings(environment)
+        (roleBindings + namespace)
+                .map { openShiftClient.performOpenShiftCommand(namespaceName, it) }
+                .forEach { responses.add(it) }
 
-        return listOf(createNamespaceResponse, updateNamespaceResponse) + updateRoleBindingsResponse
+        return responses.toList()
```

Comment:
For me the "mutability" looks wrong here. If projectRequestReponse is optional what about just replacing the forEach with .addIfNotNull(porjectRrequestResponse) and just return it?

It is not a big deal to me, so if others disagree I am fine with it.
Like in applyOpenShiftApplicationObjects